go/tools/gazelle/packages/package.go

```diff
@@ -15,7 +15,46 @@ limitations under the License.
 
 package packages
 
-import "strings"
+import (
+	"fmt"
+	"strings"
+)
+
+// PlatformConstraints is a map from config_setting labels (for example,
+// "@io_bazel_rules_go//go/platform:linux_amd64") to a sets of build tags
+// that are true on each platform (for example, "linux,amd64").
+type PlatformConstraints map[string]map[string]bool
+
+// ParsePlatformConstraints parses a string that contains all platforms and
+// tags on each platform. The string must contain an even number of fields,
+// separated by whitespace. Even fields (counting from 0) are platform labels.
```

Comment:
Per offline discussion, at this time, the platforms will not be configurable. Each platform will just be GOOS,GOARCH. We'll support platforms that both Go and Bazel work on.

We'll figure out configurability when we find a compelling use case.